output/batch/batch.go

```diff
@@ -0,0 +1,28 @@
+package batch
+
+import (
+	"io"
+	"time"
+
+	"github.com/cube2222/octosql/execution"
+)
+
+type TableFormatter func(w io.Writer, records []*execution.Record, watermark time.Time, err error) error
+
+func GetAllFields(records []*execution.Record) []string {
```

Comment:
Wouldn't it be easier and faster and more readable to create a map, then check whether fieldName exist in map if yes then continue if no then put the fieldName into map? Like that you won't need this most inner for loop (just a fairly optimal map lookup)

Reply:
And iterate through map keys to create output

Reply:
I know this is copied code but still... looks quite bad to me :/

Reply:
Oh I see now, fields must be stored in order but this won't be a problem when you store both `fields` and `fieldsMap` so that you just don't need this inner for and it won't be quadratic in time